Wikipedia/UI-V5/WMFSelfSizingWaterfallCollectionViewLayout.m

```diff
@@ -0,0 +1,14 @@
+
+#import "WMFSelfSizingWaterfallCollectionViewLayout.h"
+
+@implementation WMFSelfSizingWaterfallCollectionViewLayout
+
+- (CGSize)collectionViewContentSize {
+    if ([self.collectionView numberOfSections] == 1 && [self.collectionView numberOfItemsInSection:0] == 0) {
```

Comment:
@bgerstle Probably a little disingenuous with my subject line, this isn't the "underlying" cause… the underlying cause is the size calculation itself… which needs to be fixed. So that fix can be pushed up stream when I get around to it.

This fix here is the only way to really fix it by subclassing (without bringing it into the source). As far as I can tell, this is the only scenario it breaks on. 

But yeah… I'd like to fix it for realz and get it merged so we can just remove the subclass then.